### PR TITLE
Sync project main attributes with alternate API when we don't have a …

### PR DESCRIPTION
**Pull request by simkim**
…token

### CODE REVIEW

**Comment by simkim** [comment]
metadata can also sync without a token using archives service, but I need to support more methods to allow sync without token

**Review by andrew — CHANGES_REQUESTED** [review]
If it's not a lot of work, can we use `Faraday.get` instead of `RestClient.get`?

**Comment by simkim** [comment]
Sure

[image: Logo] <https://www.squadracer.com>

Gauthier Monserand

0668135682

Fondateur
www.squadracer.com
[image: Address-book]
<https://data.sigilium.com/signatures/rcnx0u-exdgxxefhv4ljrqfn-braarc/link>
[image: Linkedin] <https://www.linkedin.com/company/squadracer>
[image: Twitch] <https://twitch.tv/squadracer>


Le mar. 25 juin 2024 à 17:31, Andrew Nesbitt ***@***.***> a
écrit :

File: app/models/host.rb [hunk 1]
```diff
@@ -88,7 +88,11 @@ def sync_repository(full_name)
         repo.save
         repo.ping_packages_async if repo_changed && repo.persisted?
         repo.sync_extra_details_async if !repo.fork? && repo_changed && repo.persisted? && repo.files_changed?
-        repo.sync_owner
+        begin
+          repo.sync_owner
+        rescue RestClient::NotFound => e
```
Comment (andrew) [inline app/models/host.rb:93]: Can we push the error handling for `sync_owner` up into the individual host classes, as they may not all through the some error class

File: app/models/hosts/gitlab.rb [hunk 2]
```diff
@@ -1,5 +1,89 @@
...
+            yield tag
+          end
+          return if tags.count < per_page
+          page += 1
+        end
+      end
+
+      def project(id_or_full_name, license: false)
+        url = "#{@endpoint}/projects/#{CGI.escape(id_or_full_name.to_s)}?license=#{license}"
+        Rails.logger.debug("Gitlab[projec]: Fetching project from URL: #{url}")
```
Comment (andrew) [inline app/models/hosts/gitlab.rb:32]:
```suggestion
        Rails.logger.debug("Gitlab[project]: Fetching project from URL: #{url}")
```

File: openapi/api/v1/openapi.yaml [hunk 3]
```diff
@@ -15,6 +15,7 @@ externalDocs:
   url: https://github.com/ecosyste-ms/repos
 servers:
 - url: https://repos.ecosyste.ms/api/v1
+- url: https://data.code.gouv.fr/api/v1
```
Comment (andrew) [inline openapi/api/v1/openapi.yaml:18]: is this intended to be in main?

**Comment by andrew** [comment]
Just a couple of tiny changes and then will merge, looking forward to using this myself

File: openapi/api/v1/openapi.yaml (hunk 3 above)
Comment (simkim) [reply openapi/api/v1/openapi.yaml:18]: Not in this PR, but yes we would like to ask you if we could add it, to allow our users to use the API while avoiding patching before deployment. I revert on the PR.

**Comment by simkim** [comment]
The main drawback is that we have less info on user without token

**Review by andrew — APPROVED** [review]

**Comment by andrew** [comment]
🚢

**Pull request merged**